### ISSUE TITLE
"Deploy to Firebase Hosting" GitHub Action not compatible with webframeworks experiment

### ISSUE THREAD

**Issue by TomTasche** [firebase/firebase-tools#6532]
### [REQUIRED] Environment info

**firebase-tools:**
12.9.1

**Platform:**
Ubuntu
### [REQUIRED] Test case

https://github.com/TomTasche/firebase-nextjs-bug-report

### [REQUIRED] Steps to reproduce

1. Configure the above project (a basic next.js setup) for your own Firebase project
2. Run "firebase deploy" locally, wait for initial deployment
3. Create pull request to trigger the CI

### [REQUIRED] Expected behavior

CI should deploy a preview channel for the next.js project.

### [REQUIRED] Actual behavior

CI fails with the following error log: https://gist.github.com/TomTasche/aa3dc43070223f8688d36d2acb399f95

> 2023-11-18T20:07:00.0218951Z             "message": "Caller is missing permission 'iam.serviceaccounts.actAs' on service account projects/-/serviceAccounts/1022768502098-compute@developer.gserviceaccount.com.Grant the role 'roles/iam.serviceAccountUser' to the caller on the service account projects/-/serviceAccounts/1022768502098-compute@developer.gserviceaccount.com.You can do that by running 'gcloud iam service-accounts add-iam-policy-binding projects/-/serviceAccounts/1022768502098-compute@developer.gserviceaccount.com --member MEMBER --role roles/iam.serviceAccountUser'where MEMBER has a prefix like 'user:' or 'serviceAccount:'.Details and instructions for the Cloud Console can be found at https://cloud.google.com/functions/docs/reference/iam/roles#additional-configuration.",

There are a few problems with that error message:
1. the link to the documentation is wrong, should be https://cloud.google.com/functions/docs/reference/iam/roles#additional-configuration (trailing dot)
2. it does not actually tell me what to do: what am I supposed to replace "MEMBER" with?
3. it does not work: I did assign the "Service Account User" role to the mentioned service account, but I'm still experiencing the same issue:
![image](https://github.com/firebase/firebase-tools/assets/128734/9655795a-1853-41c8-8ea1-7d5956121791)

In the best case, the [next.js Firebase documentation](https://firebase.google.com/docs/hosting/frameworks/nextjs) should guide me through this step, since it seems to be mandatory for all projects anyway.

**Comment by aalej** [comment]
Hey @TomTasche, sorry to hear that you encountered this issue and thanks for filing a detailed report. Looking through the error logs, it seems like there is still a permission issue. Also, as you mentioned, there does seem to be something odd happening with the link since it includes the dot at the end, so it doesn’t redirect to the correct section of the documentation. 

To set the required permission, could you try running the command below:
```
gcloud iam service-accounts add-iam-policy-binding \
<project_number>-compute@developer.gserviceaccount.com \
--member serviceAccount:<github_action_service_account> \
--role roles/iam.serviceAccountUser
```

Where `MEMBER` is `serviceAccount:<github_action_service_account>`. Your GitHub action service account would look like `github-action-<id>@<project_id>.iam.gserviceaccount.com`. Could you check if there's still any errors being raised after running the command?

**Comment by piotrlo** [comment]
> 
> To set the required permission, could you try running the command below:
> 
> ```
> gcloud iam service-accounts add-iam-policy-binding \
> <project_number>-compute@developer.gserviceaccount.com \
> --member serviceAccount:<github_action_service_account> \
> --role roles/iam.serviceAccountUser
> ```
> 
> Where `MEMBER` is `serviceAccount:<github_action_service_account>`. Your GitHub action service account would look like `github-action-<id>@<project_id>.iam.gserviceaccount.com`. Could you check if there's still any errors being raised after running the command?

I confirm that this problem still persists and is not obvious to fix. Only this comment explaining that `github_action_service_account` should be a MEMBER helped me solve it. I was using SvelteKit.

**Comment by jeffzhkw** [comment]
I can confirm that 
```
gcloud iam service-accounts add-iam-policy-binding \
<project_number>-compute@developer.gserviceaccount.com \
--member serviceAccount:<github_action_service_account> \
--role roles/iam.serviceAccountUser
```
works for me. I ran the above command in GCloud Shell.
You may find `<project_number>` from Project Settings from Firebase Console, and `<github_action_service_account>` from Google Cloud Platform that gets redirected from Firebase "All Service Accounts"

**Comment by leoortizz** [comment]
should be fixed by #6883
